domain/src/test/java/org/oppia/android/domain/platformparameter/PlatformParameterSingletonTest.kt

```diff
@@ -0,0 +1,160 @@
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.oppia.android.app.model.PlatformParameter as ParameterValue
+import org.oppia.android.testing.robolectric.RobolectricModule
+import org.oppia.android.testing.threading.TestDispatcherModule
+import org.robolectric.annotation.Config
+import org.robolectric.annotation.LooperMode
+
+private const val STRING_PLATFORM_PARAMETER_NAME = "string_platform_parameter_name"
+private const val STRING_PLATFORM_PARAMETER_VALUE = "string_platform_parameter_value"
+
+private const val INTEGER_PLATFORM_PARAMETER_NAME = "integer_platform_parameter_name"
+private const val INTEGER_PLATFORM_PARAMETER_VALUE = 1
+
+private const val BOOLEAN_PLATFORM_PARAMETER_NAME = "boolean_platform_parameter_name"
+private const val BOOLEAN_PLATFORM_PARAMETER_VALUE = true
+
+private const val INCORRECT_PLATFORM_PARAMETER_NAME = "incorrect_platform_parameter_name"
+
+@RunWith(AndroidJUnit4::class)
+@LooperMode(LooperMode.Mode.PAUSED)
+@Config(application = PlatformParameterSingletonTest.TestApplication::class)
+class PlatformParameterSingletonTest {
+
+  @Inject
+  lateinit var platformParameterSingleton: PlatformParameterSingleton
+
+  private val mockPlatformParameterMap by lazy {
+    val stringPlatformParameter = ParameterValue.getDefaultInstance().toBuilder()
+      .setString(STRING_PLATFORM_PARAMETER_VALUE).build()
+    val integerPlatformParameter = ParameterValue.getDefaultInstance().toBuilder()
+      .setInteger(INTEGER_PLATFORM_PARAMETER_VALUE).build()
+    val booleanPlatformParameter = ParameterValue.getDefaultInstance().toBuilder()
+      .setBoolean(BOOLEAN_PLATFORM_PARAMETER_VALUE).build()
+
+    mapOf(
+      STRING_PLATFORM_PARAMETER_NAME to stringPlatformParameter,
+      INTEGER_PLATFORM_PARAMETER_NAME to integerPlatformParameter,
+      BOOLEAN_PLATFORM_PARAMETER_NAME to booleanPlatformParameter,
+    )
+  }
+
+  @Before
+  fun setUp() {
+    setUpTestApplicationComponent()
+  }
+
+  @Test
+  fun testSingleton_initPlatformParameterMap_isEmpty() {
+    assertThat(platformParameterSingleton.getPlatformParameterMap()).isEmpty()
+  }
+
+  @Test
+  fun testSingleton_initPlatformParameterMap_isNotEmpty() {
+    assertThat(platformParameterSingleton.getPlatformParameterMap().isEmpty()).isTrue()
+    platformParameterSingleton.setPlatformParameterMap(mockPlatformParameterMap)
+    assertThat(platformParameterSingleton.getPlatformParameterMap().isEmpty()).isFalse()
+  }
+
+  @Test
+  fun testSingleton_initPlatformParameterMapTwice_checkIsNotUpdatedTwice() {
+    platformParameterSingleton.setPlatformParameterMap(mockPlatformParameterMap)
+    assertThat(platformParameterSingleton.getPlatformParameterMap().isEmpty()).isFalse()
+    val emptyPlatformParameterMap = mapOf<String, ParameterValue>()
+    platformParameterSingleton.setPlatformParameterMap(emptyPlatformParameterMap)
+    assertThat(
+      platformParameterSingleton.getPlatformParameterMap()
+    ).isEqualTo(mockPlatformParameterMap)
```

Comment:
I'd also add a check that the map != empty map to guarantee nothing weird is happening with second map.

Reply:
Although if we check that the value of map is equal to mockMap that means it is not equal to emptyMap, but adding that check just to be sure